spec/factories/generic_works.rb

```diff
@@ -2,13 +2,18 @@
   factory :generic_work, aliases: [:work, :public_work], class: GenericWork do
     transient do
       user { FactoryGirl.create(:user) }
+      dates_of_work { [DateOfWork.new(start: (1900 + rand(100)).to_s)] }
     end
 
     title ['Test title']
     visibility Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PUBLIC
 
     after(:build) do |work, evaluator|
       work.apply_depositor_metadata(evaluator.user.user_key)
+
+      # unfortunately this does force saving the dates_of_work
```

Comment:
I don't think it would. All build does is give you an unsaved model, nothing magic about the models it gives you, you can create them youself too, FactoryGirl is just a convenience. 

What I discovered is if you give AF an un-saved date object and add it to the date_of_work relationship -- AF will save it at that point. Even though the 'parent' object is still not yet saved. 

FactoryGirl won't get around this. But it's not the end of the world, just another day in the life of AF. (Rails AR may do the same with associations in some cases).